src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -1010,6 +1020,63 @@ public void testXmlLoggerClosesItStreams() throws Exception {
                 0, testInfoOutputStream.getFlushCount());
     }
 
+    @Test
+    public void testDuplicatedModule() throws Exception {
+        // we need to test a module with two instances, one with id and the other not
+        final DefaultConfiguration moduleConfig1 =
+                createModuleConfig(NewlineAtEndOfFileCheck.class);
+        final DefaultConfiguration moduleConfig2 =
+                createModuleConfig(NewlineAtEndOfFileCheck.class);
+        moduleConfig2.addAttribute("id", "ModuleId");
+        final DefaultConfiguration root = new DefaultConfiguration("root");
+        root.addChild(moduleConfig1);
+        root.addChild(moduleConfig2);
+        final Checker checker = new Checker();
+        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
+        checker.configure(root);
+        // BriefUtLogger does not print the module name or id postfix,
+        // so we need to set logger manually
+        final ByteArrayOutputStream out =
+                (ByteArrayOutputStream) Whitebox.getInternalState(this, "stream");
```

Comment:
@romani I can hook in to most of the `verify` but there is some obstacles and weird things preventing a full hook because of the custom `AuditEventUtFormatter` instead of using the real `AuditEventDefaultFormatter`.

If you agree this should be fixed, lets move it to another issue.